qiskit_experiments/framework/experiment_data.py

```diff
@@ -792,9 +838,128 @@ def add_data(
                 if isinstance(datum, dict):
                     self._result_data.append(datum)
                 elif isinstance(datum, Result):
-                    self._add_result_data(datum)
+                    if datum["metadata"]:
```

Comment:
`Result` is [this object](https://github.com/Qiskit/qiskit/blob/7c89b68668b6ba74926bf767293dc8dd51ba3cfb/qiskit/result/result.py#L29) which doesn't implement `__getitem__` method.